cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -243,12 +240,12 @@ struct dispatch_t<StableAddress,
     }();
     if (!config)
     {
-      return ::cuda::std::unexpected<cudaError_t /* nvcc 12.0 with GCC 7 fails CTAD here */>(config.error());
+      return ::cuda::std::unexpected<cudaError_t /* nvcc 12.0 fails CTAD here */>(config.error());
     }
-    _CCCL_ASSERT_HOST(config->elem_per_thread > 0, "");
-    _CCCL_ASSERT_HOST(config->tile_size > 0, "");
-    _CCCL_ASSERT_HOST(config->tile_size % bulk_copy_align == 0, "");
-    _CCCL_ASSERT_HOST((sizeof...(RandomAccessIteratorsIn) == 0) != (config->smem_size != 0), ""); // logical xor
+    _CCCL_ASSERT(config->elem_per_thread > 0, "");
+    _CCCL_ASSERT(config->tile_size > 0, "");
+    _CCCL_ASSERT(config->tile_size % alignment == 0, "");
```

Comment:
Question: Here we are replacing `bulk_copy_align` with `alignment` but not everywhere, is that intentional?

Reply:
Yes, `configure_async_kernel` will be used to configure both, the `memcpy_async` kernel and the `ublkcp` kernel.